Templates/templates/CSharp/Base/EntityRequest.Base.template.tt

```diff
@@ -155,15 +155,31 @@ public string GetEntityReferencePutAsyncMethod(OdcmClass odcmClass)
     stringBuilder.Append(Environment.NewLine);
     stringBuilder.Append(@"            var objectUri = string.Format(@""{0}/users/{1}"", baseUrl, id);");
     stringBuilder.Append(Environment.NewLine);
-    stringBuilder.Append(@"            var payload = new Newtonsoft.Json.Linq.JObject(");
+    stringBuilder.Append(@"            var stream = new System.IO.MemoryStream();");
     stringBuilder.Append(Environment.NewLine);
-    stringBuilder.Append(@"                            new Newtonsoft.Json.Linq.JProperty(""@odata.id"", objectUri));");
+    stringBuilder.Append(@"            using (var writer = new System.Text.Json.Utf8JsonWriter(stream))");
+    stringBuilder.Append(Environment.NewLine);
+    stringBuilder.Append(@"            {");
+    stringBuilder.Append(Environment.NewLine);
+    stringBuilder.Append(@"                writer.WriteStartObject();");
+    stringBuilder.Append(Environment.NewLine);
+    stringBuilder.Append(@"                writer.WriteString(""@odata.id"", objectUri);");
+    stringBuilder.Append(Environment.NewLine);
+    stringBuilder.Append(@"                writer.WriteEndObject();");
+    stringBuilder.Append(Environment.NewLine);
+    stringBuilder.Append(@"                await writer.FlushAsync();");
```

Comment:
isn't flushing the writer going to flush the underlying stream as well? https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.flush?view=net-5.0#remarks

Reply:
Thanks for pointing this out. Will action and resolve.